satpy/enhancements/__init__.py

```diff
@@ -120,6 +128,65 @@ def func(band_data, index=None):
     return apply_enhancement(img.data, func, separate=True)
 
 
+def interp_scaling(
+        img: XRImage,
+        xp: ArrayLike,
+        fp: ArrayLike,
+        coordinate_divisor: Number = None,
+        **kwargs):
+    """Apply 1D linear interpolation.
+
+    This uses :func:`numpy.interp` mapped over the provided dask array chunks.
+
+    Args:
+        img: Image data to be scaled. It is assumed the data is already
+            normalized between 0 and 1.
+        xp: X coordinates of the image data points used for interpolation.
+            This is passed directly to :func:`numpy.interp`.
+        fp: Y coordinate of the output image data poitns used for
+            interpolation. This is passed directly to :func:`numpy.interp`.
+        coordinate_divisor: Divide ``xp`` and ``fp`` by this value before
+            passing using for interpolation. This is a convenience to make
+            matching normalized image data to interp coordinates or to avoid
+            floating point precision errors in YAML configuration files.
+            If not provided, ``xp`` and ``fp`` will not be modified.
```

Comment:
This docstring is very confusing. When I read it first, I was wondering why this enhancement was interpolating the data in map coordinates...
Maybe it could be rephrased a bit to make it clearer that the interpolation it done to map data values to enhanced values? maybe also not use `x` and `y`?

Reply:
Yeah I was using the terminology that `numpy` used in their docs which was probably a mistake.

Reply:
Maybe `xp` -> `input_reference` and `fp` -> `output_reference`? And instead of "coordinates" use "values".

There's also `poitns` in the `fp` explanation.

Reply:
> There's also poitns in the fp explanation.

Yeah but it is refering to the data points, not these Y coordinates.

Reply:
So what do I call `coordinate_divisor`? `reference_divisor`?

Reply:
> > There's also poitns in the fp explanation.
> 
> Yeah but it is refering to the data points, not these Y coordinates.

Sorry, I was a bit vague, and was pointing to the typo  `poitns` -> `points`.